kotlin/display-kml/src/main/java/com/esri/arcgisruntime/sample/displaykml/MainActivity.kt

```diff
@@ -0,0 +1,169 @@
+ * limitations under the License.
+ *
+ */
+
+package com.esri.arcgisruntime.sample.displaykml
+
+import android.os.Bundle
+import android.util.Log
+import android.view.Menu
+import android.view.MenuItem
+import android.view.View
+import android.widget.Toast
+import androidx.appcompat.app.AppCompatActivity
+import com.esri.arcgisruntime.layers.KmlLayer
+import com.esri.arcgisruntime.loadable.LoadStatus
+import com.esri.arcgisruntime.loadable.Loadable
+import com.esri.arcgisruntime.mapping.ArcGISMap
+import com.esri.arcgisruntime.mapping.Basemap
+import com.esri.arcgisruntime.ogc.kml.KmlDataset
+import com.esri.arcgisruntime.portal.Portal
+import com.esri.arcgisruntime.portal.PortalItem
+import kotlinx.android.synthetic.main.activity_main.*
+
+class MainActivity : AppCompatActivity() {
+
+    private val TAG: String = MainActivity::class.java.getSimpleName()
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_main)
+        progressBar.visibility = View.VISIBLE
+
+        // create a map with the dark gray canvas basemap
+        val map = ArcGISMap(Basemap.Type.DARK_GRAY_CANVAS_VECTOR, 39.0, -98.0, 4);
+        // set the map to the map view
+        mapView.map = map
+        // prompt user to make a KML source selection when the app has loaded
+        map.addDoneLoadingListener {
+            if (map.loadStatus == LoadStatus.LOADED) {
+                Toast.makeText(applicationContext, R.string.user_prompt, Toast.LENGTH_LONG).show()
+                // hide progress bar once map has loaded
+                progressBar.visibility = View.GONE
+            }
+        }
+    }
+
+    /**
+     * Shows progress indicator if the KML Layer is loading, clears all operational layers and adds
+     * the KML layer to the map as an operational layer.
+     *
+     * @param kmlLayer to add to the map
+     */
+    fun display(kmlLayer: KmlLayer) {
+        // show progress indicator if kml dataset is loading
+        progressBar.visibility = View.VISIBLE
+        kmlLayer.addDoneLoadingListener {
+            if (kmlLayer.loadStatus == LoadStatus.LOADED) {
+                progressBar.visibility = View.GONE
+            }
+        }
+        // clear operational layers before adding the KML layer to the map
+        val operationalLayers = mapView.map.operationalLayers
+        operationalLayers.clear()
+        operationalLayers.add(kmlLayer)
+    }
+
+    /**
+     * Display a KML layer from a URL.
+     */
+    fun changeSourceToURL() {
+        // create a KML data set from a URL
+        val kmlDataset = KmlDataset(getString(R.string.noaa_weather_kml_url))
+        // create a KML layer created from the KML data set and display it on the map
+        val kmlLayer = KmlLayer(kmlDataset)
+        display(kmlLayer)
+        // report layers if failed to load
+        reportErrors(kmlLayer, "Failed to load KML layer from URL: ")
```

Comment:
Is there supposed to be something after the :?

Reply:
nope, oops: just a left over remnant from a previous workflow I was trying, good spot!